### ISSUE TITLE
VSTest Task Fails even though all tests pass

### ISSUE THREAD

**Issue by camerontbelt** [microsoft/vstest#2124]
## Description
I have an azure release pipeline that deploys and runs UI tests using a VSTest task. I am now getting this issue where it fails even though all tests are passing. Please see logs below. 

## Steps to reproduce
Let me know what I can do, this is just a build pipeline task. 

## Expected behavior
I expect it to pass when all tests pass. 

## Actual behavior
Task fails even when all tests pass. 

## Diagnostic logs
**************** Completed test execution *********************
2019-08-09T17:00:33.0175876Z Test results files: C:\agent\_work\_temp\TestResults\cbelt_DC-DEVELOPMENT_2019-08-09_16_58_49_736.trx
2019-08-09T17:00:33.5028188Z Created test run: 1000376
2019-08-09T17:00:33.5030297Z Publishing test results: 10
2019-08-09T17:00:33.6097053Z Publishing test results to test run '1000376'.
2019-08-09T17:00:33.6133446Z TestResults To Publish 10, Test run id:1000376
2019-08-09T17:00:33.6233369Z Test results publishing 10, remaining: 0. Test run id: 1000376
2019-08-09T17:00:33.7778048Z Published test results: 10
2019-08-09T17:00:33.7778946Z Publishing Attachments: 1
2019-08-09T17:00:34.8993571Z ##[warning]Could not find valid build, skipping code coverage merge.
2019-08-09T17:00:35.0902244Z Completed TestExecution Model...
2019-08-09T17:00:35.1220465Z ##[warning]Vstest failed with error. Check logs for failures. There might be failed tests.
2019-08-09T17:00:35.1221743Z ##[error]Error: The process 'C:\agent\_work\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\2.155.4\Modules\DTAExecutionHost.exe' failed with exit code 1
2019-08-09T17:00:35.1226476Z ##[error]Vstest failed with error. Check logs for failures. There might be failed tests.
2019-08-09T17:00:35.1381760Z ##[section]Finishing: VsTest - Run UI Tests

## Environment
MS Server 2019 with latest Azure Dev Ops Agent

**Comment by ShreyasRmsft** [comment]
@camerontbelt could you please set system.debug = true and provide the full set of zip logs that you get.

That should provide us with enough info to get started. Thanks.

**Comment by camerontbelt** [comment]
@ShreyasRmsft 
Im not sure how to do that.

**Comment by ShreyasRmsft** [comment]
https://docs.microsoft.com/en-us/azure/devops/pipelines/troubleshooting?view=azure-devops#get-logs-to-diagnose-problems

Follow the above.v

**Comment by ShreyasRmsft** [comment]
On the Variables tab, add system.debug and set it to true. Select to allow at queue time.

Queue the build or release.

In the explorer tab, view your completed build or release and click the failing task to view its output.

If you need a copy of all the logs, click Download all logs as zip.

**Comment by camerontbelt** [comment]
[tasklog_5.log](https://github.com/microsoft/vstest/files/3497474/tasklog_5.log)

**Comment by ShreyasRmsft** [comment]
@camerontbelt could you please upload the entire zip? there are additional log files we upload that might indicate something. They'd be of the form *.diag, i would need all of them (example: Attempt--1_kkcfzd.diag)

Uploading the entire zip would help

**Comment by camerontbelt** [comment]
@ShreyasRmsft When I click on the zip link to the right of the task name in the pipeline, it just downloads the .log file. 
Unless i'm missing something.

**Comment by camerontbelt** [comment]
@ShreyasRmsft 
Ok I had to click "Download All Logs" to get that zip you mentioned. 

[ReleaseLogs_95.zip](https://github.com/microsoft/vstest/files/3498080/ReleaseLogs_95.zip)

**Comment by ShreyasRmsft** [comment]
@camerontbelt went through the logs, the issue seems extremely strange. no indications that anything went wrong. The last thing i can take a look at is the trx file, could you please upload that? Also how often is this occurring, would it be possible for you to provide a repro so that i can debug it through?

**Comment by camerontbelt** [comment]
I can try to find the trx file for you. It happens every time I run the pipeline.

**Comment by camerontbelt** [comment]
@ShreyasRmsft It tells me that I can find the file here `C:\agent\_work\_temp\TestResults\cbelt_DC-DEVELOPMENT_2019-08-13_17_08_06_300.trx` but when I navigate to that location _temp is empty.

**Comment by ShreyasRmsft** [comment]
@camerontbelt the trx gets uploaded as an attachment to your test run. Could you please navigate to the tests tab a take a look there?

Something like this

![image](https://user-images.githubusercontent.com/25194518/63145715-170c3400-c016-11e9-944a-6893dc949227.png)

**Comment by ShreyasRmsft** [comment]
@camerontbelt any updates here?

**Comment by camerontbelt** [comment]
@ShreyasRmsft I apologize. I could have sworn I uploaded the zip file to you, I guess the comment did not post. 
[ReleaseLogs_103.zip](https://github.com/microsoft/vstest/files/3521119/ReleaseLogs_103.zip)

**Comment by ShreyasRmsft** [comment]
Fixed in https://github.com/microsoft/azure-pipelines-tasks/pull/11185

**Comment by camerontbelt** [comment]
@ShreyasRmsft When will I see those changes reflected in my pipeline? I assume you guys release every two weeks.

**Comment by camerontbelt** [comment]
@ShreyasRmsft The reason I ask is because I ran it again and it still failed for the same reason even though all tests passed. But I dont know if the task has been updated or not.

**Comment by vagisha-nidhi** [comment]
@camerontbelt As @ShreyasRmsft has mentioned, the fixes have gone in, but it'll take a couple of weeks to get deployed.

**Comment by gabrieldausque** [comment]
Hello, I have the exact same issue :

**Comment by camerontbelt** [comment]
@gabrieldausque They fixed it but we're waiting on them to release the fix into production.

**Comment by gabrieldausque** [comment]
@ShreyasRmsft, @vagisha-nidhi , it seems that the fix has not been deployed. do you have any information on the day it will be ? It's quite blocking for our release now. I've tried to change to mstest to see if it is related to NUnit, but we still have the issue. Does it exists any work around on that issue ?

**Comment by camerontbelt** [comment]
@gabrieldausque The "work around" I used is for it to always pass or continue on error. It shows up as a warning now instead of an actual fail but you also have to go inspect your test logs on each release to see if any tests have failed. This just prevents from being a failed release.

**Comment by ShreyasRmsft** [comment]
@munchra could you please elaborate? Are you using the vstest task to run tests inside the container? Do you have an azure-pipelines agent setup inside the container?

**Comment by ShreyasRmsft** [comment]
@camerontbelt what version of the vstest task do you have? The fix is in the version 2.157.3. 

We do deploy every 3 weeks but the deployment is done stage wise. The inner stages get it first (mostly internal and pre-prod accounts) and then released to other stages. This fix has already been deployed to some accounts and will get deployed to the rest of the accounts by end of this week hopefully.

**Comment by camerontbelt** [comment]
@shreyasrmsft I noticed it was fixed for me late last week. It now only fails if a test fails.

**Comment by munchra** [comment]
@ShreyasRmsft We are not using Azure DevOps are are CI system. We are using Jenkins that is passing in a dotnet test command into a docker container. We are getting intermittent build failures when this command is ran, although Unit Tests are successfully ran. Extending verbosity to _detailed_ results in nothing that stands out as the cause. 

Test Run Successful.
Total tests: 131
     Passed: 131
 Total time: 1.1705 Minutes
                   Done executing task "Microsoft.TestPlatform.Build.Tasks.VSTestTask" -- FAILED. (TaskId:155)
14:05:56.596   1:7>Done building target "VSTest" in project "####.####.Tests.csproj" -- FAILED.: (TargetId:30)
14:05:56.598   1:7>Done Building Project "/src/####.####.Tests/####.####.Tests.csproj" (VSTest target(s)) -- FAILED.


That the reason for this issue was related to VSTEST and not Azure. If that's the case, and/or if needed, I can submit a separate request.

**Comment by vagisha-nidhi** [comment]
@munchra Can you please file a separate issue with full logs?
We track the issues internally as well so this messes up the tracking.
Closing this as this has been fixed.

**Comment by miroras** [comment]
Hi There.
I'm seeing the exact same issue with version 2.157.3

##[debug]Exit code 1 received from tool 'F:\Workfolder01\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\**2.157.3\**Modules\DTAExecutionHost.exe'
##[debug]STDIO streams have closed for tool 'F:\Workfolder01\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\2.157.3\Modules\DTAExecutionHost.exe'
##[warning]Vstest failed with error. Check logs for failures. There might be failed tests.
##[debug]Processed: ##vso[task.issue type=warning;]Vstest failed with error. Check logs for failures. There might be failed tests.

**Comment by miroras** [comment]
Ok - my issue was different than those mentioned above.
My issues concerned an endless looping condition which crashed the test before it was able to report a failure. and due to the fact i didn't have an updated testing platform i didn't get a dump

**Comment by diale13** [comment]
> 
> 
> @diale13, can you share you diagnostic logs (also the entirety of your build log)?

Sure:

[logs_31.zip](https://github.com/microsoft/vstest/files/7376444/logs_31.zip)

There is no hurry, just a uni job that doesent work and we cant find anything related :(
